CHANGELOG.md

```diff
@@ -14,6 +15,106 @@
 - [v0.1.0-rc2](#v010-rc2)
 - [v0.1.0-rc1](#v010-rc1)
 
+## v0.5.0
+
+API version: v1alpha2
+
+This release contains a variety of improvements, bug fixes and documentation.
+
+This release includes a large number of breaking changes in preparation for
+an upcoming `beta` release of APIs.
+
+### Breaking Changes
+
+- The `stable` release channel has been renamed to `standard`.
+  [#1182](https://github.com/kubernetes-sigs/gateway-api/pull/1182)
+- `ReferencePolicy` has been renamed to `ReferenceGrant`.
+  [#1179](https://github.com/kubernetes-sigs/gateway-api/pull/1179)
+- The `NamedAddress` value for `Gateway`'s `spec.addresses[].type` field has
+  been deprecated, and support for domain-prefixed values (like
+  `example.com/NamedAddress`) has been added instead to better represent the
+  custom nature of this support.
+  [#1178](https://github.com/kubernetes-sigs/gateway-api/pull/1178)
+- `GatewayTLSConfig`'s `CertificateRefs` field is now a slice of pointers to
+  structs instead of the structs directly.
+  [#1176](https://github.com/kubernetes-sigs/gateway-api/pull/1176)
+- Webhook validation now ensures that a path match exists when required by path
+  modifier in filter.
+  [#1171](https://github.com/kubernetes-sigs/gateway-api/pull/1171)
+- Implementations are now expected to use `404` instead of `503` responses when
+  the data-plane has no matching route.
+  [#1151](https://github.com/kubernetes-sigs/gateway-api/pull/1151)
+- The `UnsupportedExtension` named `ListenerConditionReason` has been removed.
+  [#1146](https://github.com/kubernetes-sigs/gateway-api/pull/1146)
+- The `RouteConflict` named `ListenerConditionReason` has been removed removed.
+  [#1145](https://github.com/kubernetes-sigs/gateway-api/pull/1145)
+- The `URLRewrite` filter `PreciseHostname` has replaced `Hostname`
+  [#1124](https://github.com/kubernetes-sigs/gateway-api/pull/1124)
+- Webhook validation was added to ensure that only type-appropriate fields are
+  set in `HTTPPAthModifier`.
+  [#1124](https://github.com/kubernetes-sigs/gateway-api/pull/1124)
+- `HTTPPathModifer` field `Absolute` renamed to `ReplaceFullPath`
+  [#1124](https://github.com/kubernetes-sigs/gateway-api/pull/1124)
+- `HTTPPathModifier` API replaces `Substitution` with new type-specific
+  `ReplaceFullPath` and `ReplacePrefixMatch` fields
+  [#1124](https://github.com/kubernetes-sigs/gateway-api/pull/1124)
+- Types `ConditionRouteAccepted` and `ConditionRouteResolvedRefs` are now
+  deprecated in favor of `RouteConditionAccepted` & `RouteConditionResolvedRefs`
+  [#1114](https://github.com/kubernetes-sigs/gateway-api/pull/1114)
+- The Gateway API webhook is now deployed in a `gateway-system` namespace
+  instead of `gateway-api`.
+  [#1051](https://github.com/kubernetes-sigs/gateway-api/pull/1051)
```

Comment:
Maybe this could fit in a broader "validation" section?

Reply:
Sure, done in 1e5a3b0e4cf1cb59f73cca7a06fd640b2a104412